packages/base-driver/lib/basedriver/commands/find.js

```diff
@@ -1,51 +1,108 @@
-
-helpers.findElOrElsWithProcessing = async function findElOrElsWithProcessing (strategy, selector, mult, context) {
-  this.validateLocatorStrategy(strategy);
-  try {
-    return await this.findElOrEls(strategy, selector, mult, context);
-  } catch (err) {
-    if (this.opts.printPageSourceOnFindFailure) {
-      const src = await this.getPageSource();
-      this.log.debug(`Error finding element${mult ? 's' : ''}: ${err.message}`);
-      this.log.debug(`Page source requested through 'printPageSourceOnFindFailure':`);
-      this.log.debug(src);
+/**
+ *
+ * @param {ReturnType<import('./event').EventMixin>} Base
+ * @returns {import('../driver').BaseDriverBase<import('@appium/types').TimeoutCommands & import('@appium/types').EventCommands & import('@appium/types').FindCommands>}
+ */
+export function FindMixin (Base) {
+  return class FindCommands extends Base {
+    /**
+     *
+     * @returns {Promise<Element>}
+     */
+    async findElement (strategy, selector) {
+      return await this.findElOrElsWithProcessing(strategy, selector, false);
     }
-    // still want the error to occur
-    throw err;
-  }
-};
 
-commands.findElement = async function findElement (strategy, selector) {
-  return await this.findElOrElsWithProcessing(strategy, selector, false);
-};
-
-commands.findElements = async function findElements (strategy, selector) {
-  return await this.findElOrElsWithProcessing(strategy, selector, true);
-};
+    /**
+     *
+     * @returns {Promise<Element[]>}
+     */
+    async findElements (strategy, selector) {
+      return await this.findElOrElsWithProcessing(strategy, selector, true);
+    }
 
-commands.findElementFromElement = async function findElementFromElement (strategy, selector, elementId) {
-  return await this.findElOrElsWithProcessing(strategy, selector, false, elementId);
-};
+    /**
+     *
+     * @returns {Promise<Element>}
+     */
+    async findElementFromElement (strategy, selector, elementId) {
+      return await this.findElOrElsWithProcessing(
+        strategy,
+        selector,
+        false,
+        elementId,
+      );
+    }
 
-commands.findElementsFromElement = async function findElementsFromElement (strategy, selector, elementId) {
-  return await this.findElOrElsWithProcessing(strategy, selector, true, elementId);
-};
+    /**
+     *
+     * @returns {Promise<Element[]>}
+     */
+    async findElementsFromElement (strategy, selector, elementId) {
+      return await this.findElOrElsWithProcessing(
+        strategy,
+        selector,
+        true,
+        elementId,
+      );
+    }
+    // Override the following function for your own driver, and the rest is taken
+    // care of!
+    // Returns an object which adheres to the way the JSON Wire Protocol represents elements:
+    // { ELEMENT: # }    eg: { ELEMENT: 3 }  or { ELEMENT: 1.023 }
+    /**
+     * @template {boolean} Mult
+     * @param {string} strategy
+     * @param {string} selector
+     * @param {Mult} mult
```

Comment:
mult *should* just be a boolean (whether or not we are looking for multiple elements)